gitea/repo_file.go

```diff
@@ -13,3 +13,89 @@ import (
 func (c *Client) GetFile(user, repo, ref, tree string) ([]byte, error) {
 	return c.getResponse("GET", fmt.Sprintf("/repos/%s/%s/raw/%s/%s", user, repo, ref, tree), nil, nil)
 }
+
+// IdentityOptions for a person's identity like an author or committer
+type IdentityOptions struct {
```

Comment:
Isn't there already type for this?

Reply:
@lafriks I believe the struct with the same name in `modules/file_handling/update.go` might be the one you're thinking of (moved stuff around), but needed one that is for the API to pass in that information. Didn't want to mix structures from the modules with structures in the SDK.

Reply:
Or there was another one with a slightly different name, but these structures in the SDK need to be exactly what the API call can take as the swagger is then built off of these structures.

Reply:
I think there was one with Name, Email and some other fields.

Reply:
Yes, I think you are thinking of https://github.com/go-gitea/go-sdk/blob/master/gitea/repo_commit.go#L19. A potential refactor is to extend `CommitUser` with `IdentityOptions` as a base, then CommitUser can additionally have `date`, and this file could use just IndentityOptions. I think that would decrease duplication.

Reply:
Ok, will look into doing that.

Reply:
Yeah, that looks a lot better. Didn't like putting IdentityOptions in repo_file.go since it didn't really have to do with files. Makes more sense to have it in repo_commit.go.

Reply:
Ok, made these changes and the use of them in the Gitea PR